tests/ui/external_subcommand_wrong_type.stderr

```diff
@@ -0,0 +1,8 @@
+error[E0308]: mismatched types
+  --> $DIR/external_subcommand_wrong_type.rs:13:15
+   |
+13 |     Other(Vec<CString>)
+   |               ^^^^^^^ expected struct `std::ffi::CString`, found struct `std::ffi::OsString`
+   |
+   = note: expected type `std::vec::Vec<std::ffi::CString>`
+              found type `std::vec::Vec<std::ffi::OsString>`
```

Comment:
This error message is rather unfortunate: I would very much like it to say "expected OsString", but there's nothing we can do about it 😢

Reply:
This is the best I've managed to come up with:

```
error[E0308]: mismatched types
  --> $DIR/external_subcommand_wrong_type.rs:13:15
   |
13 |     Other(Vec<CString>)
   |               ^^^^^^^ expected struct `std::ffi::CString`, found struct `std::ffi::OsString`
   |
   = note: expected type `std::vec::Vec<std::ffi::CString>`
              found type `std::vec::Vec<std::ffi::OsString>`

error[E0308]: mismatched types
  --> $DIR/external_subcommand_wrong_type.rs:13:15
   |
13 |     Other(Vec<CString>)
   |               ^^^^^^^
   |               |
   |               expected struct `std::ffi::OsString`, found struct `std::ffi::CString`
   |               this match expression has type `std::vec::Vec<std::ffi::OsString>`
   |
   = note: expected type `std::vec::Vec<std::ffi::OsString>`
              found type `std::vec::Vec<std::ffi::CString>`

```

While it is indeed possible to draw the error message we desire, I can't get rid of the first one.

Reply:
That's acceptable